docs/snippets/azure/AppHost/Program.ConfigureStorageInfra.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Azure.Provisioning.Storage;
+using Azure.Core;
+
+internal static partial class Program
+{
+    public static void ConfigureStorageInfra(IDistributedApplicationBuilder builder)
+    {
+        // <configure>
+        builder.AddAzureStorage("storage")
+            .ConfigureInfrastructure(infra =>
+            {
+                var resources = infra.GetProvisionableResources();
+
+                var storageAccount = resources.OfType<StorageAccount>()
+                                              .FirstOrDefault(r => r.BicepIdentifier is "storage")
```

Comment:
This should not look for the bicep identifier. Just use the type.

@eerhardt might disagree but I think in the 99% scenario there will be one.

Reply:
This was based on other example code, what happens when there are more than one?

Reply:
> what happens when there are more than one?

99% of the time there won't be. The only way there will be more than one is when someone else injects a `.ConfigureInfrastructure` and adds a StorageAccount to it.

You could just say `infra.GetProvisionableResources().OfType<StorageAccount>().Single();`